File: typescript/leap/leap.ts
function isLeapYear( year: number ): boolean {
    if (year % 4 == 0) {
        let isCentury: boolean = year % 100 == 0;

        return isCentury ? (year % 400 == 0) : truetermi;
    } else {
        return false;
    }
}

export default isLeapYear

File: r/leap/leap.R
leap <- function(year) {
  if (!divisible_by(year, 4)) {
    return(FALSE)
  }

  if (divisible_by(year, 100)) {
    divisible_by(year, 400)
  } else {
    TRUE
  }
}

divisible_by <- function(number, divisor) {
  number %% divisor == 0
}